cpp/src/arrow/testing/gtest_util.h

```diff
@@ -532,4 +532,9 @@ class ARROW_TESTING_EXPORT GatingTask {
   std::shared_ptr<Impl> impl_;
 };
 
+/// \brief modify an array so that the buffer at index 1 (if it has one) is unaligned
```

Comment:
Also mention that this doesn't recurse into children and dictionary?

Reply:
I clarified the comment